File: CRM.Core/Models/Biz/BizBrand.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

using CRM.Core.Interfaces;

namespace CRM.Core.Models.Biz;

/// <summary>品牌主数据（表 <c>biz_brand</c>）。</summary>
[Table("biz_brand")]
public class BizBrand : ISoftDeletable
{
    [Key]
    [Column("id")]
    public long Id { get; set; }

    /// <summary>品牌英文名。</summary>
    [StringLength(200)]
    [Column("brand_e_name")]
    public string? BrandEName { get; set; }

    /// <summary>品牌中文名。</summary>
    [StringLength(200)]
    [Column("brand_c_name")]
    public string? BrandCName { get; set; }

    /// <summary>标准品牌名（如 EN/CN 组合展示名）。</summary>
    [StringLength(300)]
    [Column("standard_brand")]
    public string? StandardBrand { get; set; }

    /// <summary>别名（可多值，逗号或换行分隔）。</summary>
    [StringLength(500)]
    [Column("alias")]
    public string? Alias { get; set; }

    /// <summary>国家/地区代码（如 CN、US）。</summary>
    [StringLength(10)]
    [Column("country_code")]
    public string? CountryCode { get; set; }

    /// <summary>国家/地区名称。</summary>
    [StringLength(100)]
    [Column("country")]
    public string? Country { get; set; }

    /// <summary>备注。</summary>
    [StringLength(500)]
    [Column("remark")]
    public string? Remark { get; set; }

    /// <summary>创建人用户 ID。</summary>
    [StringLength(36)]
    [Column("create_by_user_id")]
    public string? CreateByUserId { get; set; }

    /// <summary>创建日期。</summary>
    [Column("create_time")]
    public DateTime? CreateTime { get; set; }

    /// <summary>审核状态：1 待审核，2 已审核。</summary>
    [Column("audit_status")]
    public short? AuditStatus { get; set; }

    /// <summary>审核人用户 ID。</summary>
    [StringLength(36)]
    [Column("audit_by_user_id")]
    public string? AuditByUserId { get; set; }

    /// <summary>审核日期。</summary>
    [Column("audit_time")]
    public DateTime? AuditTime { get; set; }

    /// <summary>是否已删除（软删除）。</summary>
    [Column("is_deleted")]
    public bool IsDeleted { get; set; }

    /// <summary>删除时间。</summary>
    [Column("deleted_at")]
    public DateTime? DeletedAt { get; set; }

    /// <summary>删除操作人用户 ID。</summary>
    [StringLength(36)]
    [Column("deleted_by_user_id")]
    public string? DeletedByUserId { get; set; }
}

File: CRM.Web/src/api/bizBrand.ts
import apiClient from './client'

export type BizBrandRow = {
  id: number
  brandEName?: string | null
  brandCName?: string | null
  standardBrand?: string | null
  alias?: string | null
  countryCode?: string | null
  country?: string | null
  remark?: string | null
  createByUserId?: string | null
  createUserName?: string | null
  createTime?: string | null
  auditStatus?: number | null
  auditByUserId?: string | null
  auditUserName?: string | null
  auditTime?: string | null
}

export type BizBrandPaged = {
  items: BizBrandRow[]
  total: number
}

export type UpsertBizBrandPayload = {
  brandEName?: string | null
  brandCName?: string | null
  standardBrand?: string | null
  alias?: string | null
  countryCode?: string | null
  country?: string | null
  remark?: string | null
}

/** 品牌下拉选项（RFQ 等场景） */
export type BizBrandOption = {
  id: number
  standardBrand?: string | null
  auditStatus?: number | null
  brandEName?: string | null
  brandCName?: string | null
  alias?: string | null
}

function normalizeBizBrandOption(raw: Record<string, unknown>): BizBrandOption {
  const id = Number(raw.id ?? raw.Id ?? 0)
  return {
    id: Number.isFinite(id) && id > 0 ? id : 0,
    standardBrand: pickString(raw.standardBrand ?? raw.StandardBrand),
    auditStatus: pickNumber(raw.auditStatus ?? raw.AuditStatus),
    brandEName: pickString(raw.brandEName ?? raw.BrandEName),
    brandCName: pickString(raw.brandCName ?? raw.BrandCName),
    alias: pickString(raw.alias ?? raw.Alias)
  }
}

export function bizBrandRowToOption(row: BizBrandRow): BizBrandOption {
  return {
    id: row.id,
    standardBrand: row.standardBrand,
    auditStatus: row.auditStatus,
    brandEName: row.brandEName,
    brandCName: row.brandCName
  }
}

const BASE = '/api/v1/biz-brands'

function pickString(v: unknown): string | null | undefined {
  if (v == null) return v as null | undefined
  if (typeof v === 'string') return v
  if (typeof v === 'number' || typeof v === 'boolean') return String(v)
  return null
}

function pickNumber(v: unknown): number | null | undefined {
  if (v == null || v === '') return v === '' ? null : (v as null | undefined)
  const n = Number(v)
  return Number.isFinite(n) ? n : null
}

function normalizeBizBrandRow(raw: Record<string, unknown>): BizBrandRow {
  return {
    id: Number(raw.id ?? raw.Id ?? 0),
    brandEName: pickString(raw.brandEName ?? raw.BrandEName),
    brandCName: pickString(raw.brandCName ?? raw.BrandCName),
    standardBrand: pickString(raw.standardBrand ?? raw.StandardBrand),
    alias: pickString(raw.alias ?? raw.Alias),
    countryCode: pickString(raw.countryCode ?? raw.CountryCode),
    country: pickString(raw.country ?? raw.Country),
    remark: pickString(raw.remark ?? raw.Remark),
    createByUserId: pickString(raw.createByUserId ?? raw.CreateByUserId),
    createUserName: pickString(raw.createUserName ?? raw.CreateUserName),
    createTime: pickString(raw.createTime ?? raw.CreateTime),
    auditStatus: pickNumber(raw.auditStatus ?? raw.AuditStatus),
    auditByUserId: pickString(raw.auditByUserId ?? raw.AuditByUserId),
    auditUserName: pickString(raw.auditUserName ?? raw.AuditUserName),
    auditTime: pickString(raw.auditTime ?? raw.AuditTime)
  }
}

export const bizBrandApi = {
  async fetchList(params: {
    brandCName?: string
    brandEName?: string
    standardBrand?: string
    alias?: string
    country?: string
    remark?: string
    auditStatus?: number
    createTimeFrom?: string
    createTimeTo?: string
    exactMatch?: boolean
    page?: number
    pageSize?: number
  }): Promise<BizBrandPaged> {
    const raw = await apiClient.get<Record<string, unknown>>(BASE, {
      params: {
        brandCName: params.brandCName?.trim() || undefined,
        brandEName: params.brandEName?.trim() || undefined,
        standardBrand: params.standardBrand?.trim() || undefined,
        alias: params.alias?.trim() || undefined,
        country: params.country?.trim() || undefined,
        remark: params.remark?.trim() || undefined,
        auditStatus: params.auditStatus ?? undefined,
        createTimeFrom: params.createTimeFrom || undefined,
        createTimeTo: params.createTimeTo || undefined,
        exactMatch: params.exactMatch ? true : undefined,
        page: params.page ?? 1,
        pageSize: params.pageSize ?? 20
      }
    })
    const itemsRaw = (raw.items ?? raw.Items) as Record<string, unknown>[] | undefined
    return {
      items: (itemsRaw ?? []).map((row) => normalizeBizBrandRow(row)),
      total: Number(raw.total ?? raw.Total ?? 0)
    }
  },

  async update(id: number, body: UpsertBizBrandPayload): Promise<BizBrandRow> {
    const raw = await apiClient.put<Record<string, unknown>>(`${BASE}/${id}`, body)
    return normalizeBizBrandRow(raw)
  },

  async create(body: UpsertBizBrandPayload): Promise<BizBrandRow> {
    const raw = await apiClient.post<Record<string, unknown>>(BASE, body)
    return normalizeBizBrandRow(raw)
  },

  async remove(id: number): Promise<void> {
    await apiClient.delete(`${BASE}/${id}`)
  },

  async approve(id: number): Promise<BizBrandRow> {
    const raw = await apiClient.post<Record<string, unknown>>(`${BASE}/${id}/approve`, {})
    return normalizeBizBrandRow(raw)
  },

  async fetchOptions(params?: { keyword?: string; pageSize?: number }): Promise<BizBrandOption[]> {
    const raw = await apiClient.get<Record<string, unknown>[] | Record<string, unknown>>(`${BASE}/options`, {
      params: {
        keyword: params?.keyword?.trim() || undefined,
        pageSize: params?.pageSize ?? 50
      }
    })
    const list = Array.isArray(raw) ? raw : ((raw.items ?? raw.Items) as Record<string, unknown>[] | undefined)
    return (list ?? []).map((row) => normalizeBizBrandOption(row))
  },

  async getById(id: number): Promise<BizBrandRow> {
    const raw = await apiClient.get<Record<string, unknown>>(`${BASE}/${id}`)
    return normalizeBizBrandRow(raw)
  },

  async rememberLearnedMapping(body: { sourceText: string; brandId: number }): Promise<void> {
    await apiClient.post(`${BASE}/learned-mappings/remember`, body)
  },

  async resolveLearnedMappings(body: { sourceTexts: string[] }): Promise<
    Array<{ sourceText: string; sourceKey: string; brandId: number; standardBrand?: string | null }>
  > {
    const raw = await apiClient.post<
      Array<Record<string, unknown>> | Record<string, unknown>
    >(`${BASE}/learned-mappings/resolve`, body)
    const list = Array.isArray(raw) ? raw : ((raw.items ?? raw.Items) as Record<string, unknown>[] | undefined)
    return (list ?? []).map((row) => ({
      sourceText: String(row.sourceText ?? row.SourceText ?? ''),
      sourceKey: String(row.sourceKey ?? row.SourceKey ?? ''),
      brandId: Number(row.brandId ?? row.BrandId ?? 0),
      standardBrand: pickString(row.standardBrand ?? row.StandardBrand)
    }))
  }
}
